engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -249,15 +252,10 @@ private void initRenderGraph() {
         renderGraph.add(bloomPassesNode);
         renderGraph.add(blurPassesNode);
         renderGraph.add(finalPostProcessingNode);
-    }
 
-    private static <T extends Node> T createInstance(Class<T> type, Context context) {
-        // Attempt constructor-based injection first
-        T node = InjectionHelper.createWithConstructorInjection(type, context);
-        // Then fill @In fields
-        InjectionHelper.inject(node, context);
-        node.initialise();
-        return type.cast(node);
+        RenderPipelineGenerator generator = new RenderPipelineGenerator(renderGraph);
+        RenderPipelineOptimizer optimizer = new RenderPipelineOptimizer(generator.generate());
+        renderPipelineExecutor = new RenderPipelineExecutor(optimizer.optimize());
```

Comment:
I think `renderTaskList` is a better name.  Also how about:

```ruby
intermediatePipeline = IntermediatePipelineGenerator.createPipelineFrom(renderGraph);
renderTaskList = RenderPipelineGenerator.createPipelineFrom(intermediatePipeline);
```
`IntermediatePipelineGenerator.createPipelineFrom` can call `renderGraph`s `getNodesInTopologicalOrder`? I don't see any usage at this point of having `processableNodes` as a separate variable. 

In addition to a possible naming issue is having `intermediatePipeline` and a `renderablePipeline` as a variable, seems like both are different parts of the pipeline. 

Another idea that I think would be useful for the architecture to have `RenderPipelineOptimizer` inside a `RenderTaskListGenerator` (by replacing `RenderPipelineGenerator class) class. In that way we wouldn't concerned about people thinking that pre-optimized pipeline would work, because there will be no way to access it. Generated output will be a `renderTaskList` ready for fast execution. So it would be just:

```java
renderTaskList = RenderTaskListGenerator.createFrom(renderGraph);
```

👍 for converting both of `RenderPipeline` and `OptimizedPipeline` into `List`s.